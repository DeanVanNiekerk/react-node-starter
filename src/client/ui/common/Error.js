import React from 'react'

const Error = ({ text = "An error has occured, please reload the application" }) => {

    return <div className="mt-5 mb-5">
        <div className="row justify-content-center text-danger">
            <h5>{text}</h5>
        </div>
    </div>

}

export default Error