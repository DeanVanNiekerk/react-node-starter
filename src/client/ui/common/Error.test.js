import React from 'react';
import Error from './Error';
import renderer from 'react-test-renderer';

describe('Error', () => {

    it('renders correctly - with no text', () => {

        const tree = renderer
            .create(<Error />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders correctly - with text', () => {

        const tree = renderer
            .create(<Error text="Custom error message" />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

});