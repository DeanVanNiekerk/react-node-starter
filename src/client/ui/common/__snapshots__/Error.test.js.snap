// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders correctly - with no text 1`] = `
<div
  className="mt-5 mb-5"
>
  <div
    className="row justify-content-center text-danger"
  >
    <h5>
      An error has occured, please reload the application
    </h5>
  </div>
</div>
`;

exports[`Error renders correctly - with text 1`] = `
<div
  className="mt-5 mb-5"
>
  <div
    className="row justify-content-center text-danger"
  >
    <h5>
      Custom error message
    </h5>
  </div>
</div>
`;
